src/dapps/SecureTransaction/SecureTransaction.vue

```diff
@@ -0,0 +1,460 @@
+        this.amount <= parseFloat(this.parsedBalance, 10);
+    },
+    parsedBalance(newVal) {
+      this.parsedBalance = newVal;
+    },
+    gasAmount(newVal) {
+      this.gasAmount = newVal;
+      if (this.verifyAddr()) {
+        this.estimateGas();
+      }
+      this.$store.dispatch('setGasPrice', Number(newVal));
+    },
+    selectedCurrency(newVal) {
+      this.selectedCurrency = newVal;
+      this.estimateGas();
+    }
+  },
+  mounted() {
+    const address = this.$store.state.wallet.getAddressString();
+
+    this.web3.eth
+      .getBalance(address)
+      .then(res => {
+        this.balance = this.web3.utils.fromWei(res, 'ether');
+        this.$store.dispatch('setAccountBalance', this.balance);
+        this.parsedBalance = this.balance;
+        this.gasAmount = this.gasPrice;
+      })
+      .catch(err => {
+        // eslint-disable-next-line no-console
+        console.error(err);
+      });
+    this.web3.eth.net
+      .getId()
+      .then(res => {
+        this.networkID = res;
+        this.validNetwork = this.networkID === CoralConfig.chainID;
+      })
+      .catch(err => {
+        // eslint-disable-next-line no-console
+        console.error(err);
+      });
+  },
+  methods: {
+    debouncedAmount: utils._.debounce(function(e) {
+      this.amount = parseFloat(
+        new BigNumber(e.target.value).decimalPlaces(18).toFixed(),
+        10
+      );
+      this.getSafeSendFee();
+      if (this.verifyAddr()) {
+        this.estimateGas();
+      }
+    }, 501),
+    debounceInput: utils._.debounce(function(e) {
+      this.address = normalise(e.target.value);
+    }, 110),
+    copyToClipboard(ref) {
+      this.$refs[ref].select();
+      document.execCommand('copy');
+    },
+    async createTx() {
+      this.nonce = await this.$store.state.web3.eth.getTransactionCount(
+        this.wallet.getAddressString(),
+        'latest'
+      );
+      const value = this.amount === '' ? 0 : unit.toWei(this.amount, 'ether');
+      const CoralSafeSendContract = new this.$store.state.web3.eth.Contract(
+        CoralConfig.safeSendEscrowContractAbi,
+        CoralConfig.safeSendEscrowContractAddress
+      );
+      const to =
+        this.resolvedAddress !== '' ? this.resolvedAddress : this.address;
+      const protectionLevel = 20;
+      const query = CoralSafeSendContract.methods['deposit'](
+        to,
+        protectionLevel
+      );
+      const encodedABI = query.encodeABI();
+      const gasLimit = CoralConfig.gasLimitSuggestion;
+      const valueLessGas =
+        parseFloat(value, 10) - Number(unit.toWei(gasLimit, 'gwei'));
+      this.raw = {
+        from: this.$store.state.wallet.getAddressString(),
+        value: valueLessGas,
+        to: CoralConfig.safeSendEscrowContractAddress,
+        nonce: this.nonce,
+        gas: gasLimit,
+        data: encodedABI,
+        gasPrice: Number(unit.toWei(this.$store.state.gasPrice, 'gwei')),
```

Comment:
use `this.gasPrice` since it's imported already

Reply:
also do `new BigNumber(unit.toWei(this.gasPrice, 'gwei')).toString()`